sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CodeGenerationSuite.scala

```diff
@@ -113,7 +113,7 @@ class CodeGenerationSuite extends SparkFunSuite with ExpressionEvalHelper {
     assert(actual.length == 1)
     val expected = UTF8String.fromString("abc")
 
-    if (!checkResult(actual.head, expected, expressions.head.dataType)) {
+    if (!checkResult(actual.head, expected, expressions.head.dataType, expressions.head.nullable)) {
```

Comment:
It's a little weird to ask the caller to provide both `expected` and `exprNullable `, and then use `exprNullable ` to validate `expected.` Can we set a default value for `exprNullable` in `checkResult`?

Reply:
That is another option that I thought. On the other hand, to set default has a risk to overlook a possible incosistency between value and `nullable` at top level of `expected`.

Do we use the default value at the all of callers of `checkResult`?

Reply:
maybe we should provide an overload of `checkResult` that takes `Expression`.